tests/test_uploads.py

```diff
@@ -85,3 +85,72 @@ def test_cli_upload(monkeypatch):
          'test-data',
          'tests/twopoints.geojson'])
     assert result.exit_code == 0
+
+
+@responses.activate
+def test_cli_upload_unknown_error(monkeypatch):
+
+    monkeypatch.setattr(mapbox.services.uploads, 'boto3_session', MockSession)
+
+    # Credentials
+    query_body = """
+       {{"key": "_pending/{username}/key.test",
+         "accessKeyId": "ak.test",
+         "bucket": "tilestream-tilesets-production",
+         "url": "https://tilestream-tilesets-production.s3.amazonaws.com/_pending/{username}/key.test",
+         "secretAccessKey": "sak.test",
+         "sessionToken": "st.test"}}""".format(username=username)
+
+    responses.add(
+        responses.GET,
+        'https://api.mapbox.com/uploads/v1/{0}/credentials?access_token={1}'.format(username, access_token),
+        match_querystring=True,
+        body=query_body, status=200,
+        content_type='application/json')
+
+    responses.add(
+        responses.POST,
+        'https://api.mapbox.com/uploads/v1/{0}?access_token={1}'.format(username, access_token),
+        match_querystring=True,
+        body='{"message":"Something went wrong"}', status=500,
+        content_type='application/json')
+
+    runner = CliRunner()
+    result = runner.invoke(
+        main_group,
+        ['--access-token', access_token,
+         'upload',
+         'test-data',
+         'tests/twopoints.geojson'])
+    assert result.exit_code == 1
+    assert result.output == 'Error: {"message":"Something went wrong"}\n'
+
+@responses.activate
+def test_cli_upload_doesnotexist(monkeypatch):
+
+    monkeypatch.setattr(mapbox.services.uploads, 'boto3_session', MockSession)
+
+    # Credentials
+    query_body = """
+       {{"key": "_pending/{username}/key.test",
+         "accessKeyId": "ak.test",
+         "bucket": "tilestream-tilesets-production",
+         "url": "https://tilestream-tilesets-production.s3.amazonaws.com/_pending/{username}/key.test",
+         "secretAccessKey": "sak.test",
+         "sessionToken": "st.test"}}""".format(username=username)
+
+    responses.add(
+        responses.GET,
+        'https://api.mapbox.com/uploads/v1/{0}/credentials?access_token={1}'.format(username, access_token),
+        match_querystring=True,
+        body=query_body, status=200,
+        content_type='application/json')
```

Comment:
Maybe not the right place to mention this, but I've been thinking a little about whether and how to make these responses into test fixtures. I guess as soon as we begin to repeat fake responses, we should turn them into a fixture?